### PR TITLE
Removed VOKMigrationFailureOptionWipeRecoveryAndAlert, take 2

### PR DESCRIPTION
**Pull request by brockboland**
Simplified version of #104 

- Removed `VOKMigrationFailureOptionWipeRecoveryAndAlert`
- Added `VOKMigrationFailureWipeRecoveryNotificationName`
- Added a test for this
- Updated the CocoaPods version used
- Removed `VOK_TARGET_USES_UIKIT`

@vokal/ios-developers @chillpop apologies for this: @designatednerd was right, there was no need for a separate project to test this change.

### CODE REVIEW

**Comment by brockboland** [comment]
Oh, @designatednerd, I also addressed your comments in the tests

File: Pod/Classes/VOKCoreDataManager.m [hunk 1]
```diff
@@ -175,46 +177,18 @@ - (void)addPersistentStoreToCoordinator:(NSPersistentStoreCoordinator *)persiste
                                                         options:options
                                                           error:&error]) {
         switch (self.migrationFailureOptions) {
-            case VOKMigrationFailureOptionWipeRecoveryAndAlert:
-#if VOK_TARGET_USES_UIKIT
```
Comment (chillpop) [inline Pod/Classes/VOKCoreDataManager.m]: I think this macro can be deleted since it was only used here

Comment (brockboland) [inline Pod/Classes/VOKCoreDataManager.m]: Good point!

File: Pod/Classes/VOKCoreDataManager.h [hunk 2]
```diff
@@ -25,10 +25,14 @@ typedef NS_ENUM (NSInteger, VOKMigrationFailureOption) {
     VOKMigrationFailureOptionNone,
     /// Silently delete and recreate the database file, data will be erased, but instability and crashing will be avoided
     VOKMigrationFailureOptionWipeRecovery,
-    /// Same as VOKMigrationFailureOptionWipeRecoveryAndAlert, but will also notify the user that data has been erased via an on screen alert. NOTE: not yet implemented on OS X
-    VOKMigrationFailureOptionWipeRecoveryAndAlert,
 };
 
+/**
+ * Name of the NSNotification that's posted after a successful DB wipe following a migration failure.
```
Comment (chillpop) [inline Pod/Classes/VOKCoreDataManager.h]: If you're adding this notification on success, worth adding one for failure too?

Comment (brockboland) [inline Pod/Classes/VOKCoreDataManager.h]: It already raises an `NSException` in that case, so I think we're good

Comment (chillpop) [inline Pod/Classes/VOKCoreDataManager.h]: oh, I see, I was misreading the comment.  This is exactly what I was expecting

**Pull request merged**